### PR TITLE
Migrate 'GeocoderWidget' component from makeStyles to styled-component

### PR DESCRIPTION
**Pull request by aaranadev**
# Description

This PR includes a refactorization of GeocoderWidget component, all classes from makestyles are been converted to styled components

Shortcut: [297037](https://app.shortcut.com/cartoteam/story/297037/migrate-away-from-makestyles-i-widgets)

## Type of change

- Refactor

# Acceptance

The code resultant must returns same styles component visualization as the same before changes

1. Overview a general visualization from v1.x to v2.x

### CODE REVIEW

**Review by vmilan — COMMENTED** [review]
I was able to visualize the component (in a very hacky way) in Storybook, 
just the raw UI elements... Looking fairly good:

Screenshot 2023-04-13 at 18 27 40
Screenshot 2023-04-13 at 18 27 09

Screenshots with my suggestions applied

File: packages/react-widgets/src/widgets/GeocoderWidget.js [hunk 1]
```diff
@@ -4,32 +4,28 @@ import { PropTypes } from 'prop-types';
...
-  input: {
-    ...theme.typography.body2,
-    width: `calc(100% - ${theme.spacing(5)})`,
-    marginLeft: theme.spacing(1)
-  }
+const StyledPaper = styled(Paper)(({ theme: { spacing } }) => ({
+  height: spacing(4.5),
+  width: spacing(30),
+  paddingLeft: spacing(1.5),
+  borderRadius: 4,
```
[the remark this answers is not in the document]
Comment (aaranadev) [reply packages/react-widgets/src/widgets/GeocoderWidget.js:14]: Done

File: packages/react-widgets/src/widgets/GeocoderWidget.js [hunk 2]
```diff
@@ -78,27 +73,30 @@ function GeocoderWidget(props = {}) {
   }, [geocoderResult, dispatch]);
 
   return (
-    <Paper className={`${props.className} ${classes.paperInput}`} elevation={2}>
+    <StyledPaper className={`${props.className}`} elevation={2}>
       {loading ? (
-        <CircularProgress size={20} className={classes.icon} />
+        <StyledIcon>
+          <CircularProgress size={20} />
```
[the remark this answers is not in the document]
Comment (aaranadev) [reply packages/react-widgets/src/widgets/GeocoderWidget.js:79]: Done

File: packages/react-widgets/src/widgets/GeocoderWidget.js [hunk 3]
```diff
@@ -4,32 +4,28 @@ import { PropTypes } from 'prop-types';
...
+  width: spacing(30),
+  paddingLeft: spacing(1.5),
+  borderRadius: 4,
+  display: 'flex',
+  alignItems: 'center'
+}));
+
+const StyledIcon = styled(SvgIcon)(({ theme: { palette, typography } }) => ({
+  fill: palette.text.secondary,
+  fontSize: `${typography.body2.lineHeight}em`,
```
[the remark this answers is not in the document]
Comment (aaranadev) [reply packages/react-widgets/src/widgets/GeocoderWidget.js:21]: Done

File: packages/react-widgets/src/widgets/GeocoderWidget.js [hunk 4]
```diff
@@ -4,32 +4,28 @@ import { PropTypes } from 'prop-types';
...
+const StyledPaper = styled(Paper)(({ theme: { spacing } }) => ({
+  height: spacing(4.5),
+  width: spacing(30),
+  paddingLeft: spacing(1.5),
+  borderRadius: 4,
+  display: 'flex',
+  alignItems: 'center'
+}));
+
+const StyledIcon = styled(SvgIcon)(({ theme: { palette, typography } }) => ({
```
[the remark this answers is not in the document]
Comment (aaranadev) [reply packages/react-widgets/src/widgets/GeocoderWidget.js:19]: I've changed the styled component by `sx` props with fill so it's de unique style applied.

File: packages/react-widgets/src/widgets/GeocoderWidget.js [hunk 5]
```diff
@@ -78,27 +73,30 @@ function GeocoderWidget(props = {}) {
   }, [geocoderResult, dispatch]);
 
   return (
-    <Paper className={`${props.className} ${classes.paperInput}`} elevation={2}>
+    <StyledPaper className={`${props.className}`} elevation={2}>
       {loading ? (
-        <CircularProgress size={20} className={classes.icon} />
+        <StyledIcon>
```
[the remark this answers is not in the document]
Comment (aaranadev) [reply packages/react-widgets/src/widgets/GeocoderWidget.js:78]: I've removed it

File: packages/react-widgets/src/widgets/GeocoderWidget.js [hunk 6]
```diff
@@ -4,32 +4,28 @@ import { PropTypes } from 'prop-types';
...
+  alignItems: 'center'
+}));
+
+const StyledIcon = styled(SvgIcon)(({ theme: { palette, typography } }) => ({
+  fill: palette.text.secondary,
+  fontSize: `${typography.body2.lineHeight}em`,
+  height: '1em'
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme: { typography, spacing } }) => ({
```
[the remark this answers is not in the document]
Comment (aaranadev) [reply packages/react-widgets/src/widgets/GeocoderWidget.js:25]: Done

File: packages/react-widgets/src/widgets/GeocoderWidget.js [hunk 7]
```diff
@@ -4,32 +4,28 @@ import { PropTypes } from 'prop-types';
...
+  height: spacing(4.5),
+  width: spacing(30),
+  paddingLeft: spacing(1.5),
+  borderRadius: 4,
+  display: 'flex',
+  alignItems: 'center'
+}));
+
+const StyledIcon = styled(SvgIcon)(({ theme: { palette, typography } }) => ({
+  fill: palette.text.secondary,
```
[the remark this answers is not in the document]
Comment (aaranadev) [reply packages/react-widgets/src/widgets/GeocoderWidget.js:20]: I've used `sx` props directly

File: packages/react-widgets/src/widgets/GeocoderWidget.js [hunk 8]
```diff
@@ -4,32 +4,28 @@ import { PropTypes } from 'prop-types';
...
+  fontSize: `${typography.body2.lineHeight}em`,
+  height: '1em'
```
[the remark this answers is not in the document]
Comment (aaranadev) [reply packages/react-widgets/src/widgets/GeocoderWidget.js:22]: It's unnecessary if I removed the `SVGIcon`

**Pull request merged**